python/xoscar/backends/communication/socket.py

```diff
@@ -258,12 +259,24 @@ async def connect(
         return SocketClient(local_address, dest_address, channel)
 
 
-TEMPDIR = tempfile.gettempdir()
+def _get_or_create_default_unix_socket_dir():
+    import sys
+
+    if sys.platform.startswith("win"):
+        unix_socket_dir = XOSCAR_UNIX_SOCKET_DIR_WIN
+    else:
+        unix_socket_dir = XOSCAR_UNIX_SOCKET_DIR
+    os.makedirs(unix_socket_dir, exist_ok=True)
+    os.chmod(unix_socket_dir, mode=0o777)
```

Comment:
Is it possible that the user that runs the script does not have the permission to run this?

Reply:
Good point. How about catch the possible permission err and print a warning?